target_chains/starknet/contracts/src/pyth.cairo

```diff
@@ -229,6 +255,23 @@ mod pyth {
             self.get_total_fee(num_updates)
         }
 
+        fn update_price_feeds_if_necessary(
+            ref self: ContractState,
+            update: ByteArray,
+            required_publish_times: Array<PriceFeedPublishTime>
+        ) {
+            let mut i = 0;
+            while i < required_publish_times.len() {
+                let item = required_publish_times.at(i);
+                let latest_time = self.latest_price_info.read(*item.price_id).publish_time;
+                if latest_time < *item.publish_time {
+                    self.update_price_feeds(update);
+                    break;
+                }
+                i += 1;
+            }
```

Comment:
in the evm implementation we revert here if it's unsuccessful. I think would be nice to have it here too. The rationale is that if we revert, there is a chance that it gets reverted on simulation and then users won't pay any fees.

Reply:
Fixed.